airflow/models/variable.py

```diff
@@ -182,7 +181,6 @@ def set(
 
     @staticmethod
     @provide_session
-    @internal_api_call
```

Comment:
I assume the change you make is the "simplest way" to remove it. Nevertheless one main problem of AIP-44 was that the core needed to be re-structured and is now a bit cluttered. Example here as well: The original `set()` method is just calling the internal api method `_set()`. That was just introduced to keep interface compatibility but have the DB access separation layer.

Can you check how it was implemented prior AIP-44 and rather revert it to this state back? Means that no `_set()` but the `set()` directly carries-out the call and makes the DB execution.

Reply:
Ah, just barely remembered, it was even me who implemented AIP-44 in this class in #41414

Reply:
Yeah. We should merge those methods.

Reply:
We can merge it as follow up :)